extensions/binding/org.eclipse.smarthome.binding.astro.test/src/test/groovy/org/eclipse/smarthome/binding/astro/handler/test/AstroCommandTest.groovy

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.astro.handler.test
+
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.*
+
+import org.eclipse.smarthome.config.core.Configuration
+import org.eclipse.smarthome.core.events.Event
+import org.eclipse.smarthome.core.events.EventSubscriber
+import org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber
+import org.eclipse.smarthome.core.thing.ChannelUID
+import org.eclipse.smarthome.core.thing.ThingStatus
+import org.eclipse.smarthome.core.types.RefreshType
+import org.eclipse.smarthome.core.types.State
+import org.junit.Test
+import org.eclipse.smarthome.binding.astro.AstroBindingConstants
+import org.eclipse.smarthome.binding.astro.test.AstroOSGiTest
+import org.eclipse.smarthome.binding.astro.test.AstroOSGiTest.AcceptedItemType
+
+/**
+ * OSGi test for the {@link AstroThingHandler}
+ * <p>
+ * This class tests the commands for the astro thing.
+ *
+ * @author Petar Valchev - Initial Implementation
+ * @author Svilen Valkanov - Added missing configuration parameters and replaced headers
+ *
+ */
+class AstroCommandTest extends AstroOSGiTest {
+    // a listener for item state update events
+    private EventSubscriberMock eventSubscriberMock
+
+    @Test
+    public void 'refresh command updates the state of the channels'(){
+        Configuration thingConfiguration = new Configuration()
+        thingConfiguration.put(GEOLOCATION_PROPERTY, GEOLOCATION_VALUE)
+        thingConfiguration.put(INTERVAL_PROPERTY, INTERVAL_DEFAULT_VALUE)
+        initialize(TEST_SUN_THING_ID, DEFAULT_TEST_CHANNEL_ID, AcceptedItemType.DATE_TIME, thingConfiguration)
+
+        waitForAssert({
+            assertThat "The status of the thing $astroThing was not as expected",
+                    astroThing.getStatus(),
+                    is(equalTo(ThingStatus.ONLINE))
+        })
+
+        eventSubscriberMock = new EventSubscriberMock()
+        registerService(eventSubscriberMock, EventSubscriber.class.getName())
+
+        ChannelUID testItemChannelUID = getChannelUID(DEFAULT_TEST_CHANNEL_ID)
+
+        astroHandler.handleCommand(testItemChannelUID, RefreshType.REFRESH)
+        waitForAssert({
+            assertThat "An event for update of channel was not received",
```

Comment:
I know, you want to be nice an provide an error message in case the assertion failed. However, IMHO this is very confusing and is a flaw in their API: The hamcrest matchers are _intentionally_ made in a way to produce _readable code_, e.g.

```
assertThat eventReceived, is(true)
```

so you can almost read it like natural language. 

However, in your attempt to provide some even better human readable language, you broke this concept completely:

```
assertThat "An event for update of channel was not received", //...
```

It's the exact opposite of what you actually want to assure. 

Therefore: please remove all of these strings and try to make it a crisp one-liner of code instead.